SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -458,7 +459,308 @@ TMap<Worker_ComponentId, FString> CreateComponentIdToClassPathMap()
+						break;
+					case SCHEMA_OwnerOnly:
+						Writer.Printf("unreal.generated.{0}.{1}OwnerOnly,", ActorClassName.ToLower(), ActorClassName);
+						break;
+					case SCHEMA_Handover:
+						Writer.Printf("unreal.generated.{0}.{1}Handover,", ActorClassName.ToLower(), ActorClassName);
+						break;
+					default:
+						break;
+					}
+				}
+			});
+			// Actor static subobjects.
+			for (const auto& ActorSubObjectData : GeneratedActorClass.Value.SubobjectData)
+			{
+				const FString ActorSubObjectName = UnrealNameToSchemaComponentName(ActorSubObjectData.Value.Name.ToString());
+				ForAllSchemaComponentTypes([&](ESchemaComponentType SchemaType) {
+					const Worker_ComponentId& ComponentId = ActorSubObjectData.Value.SchemaComponents[SchemaType];
+					if (ComponentId != 0)
+					{
+						switch (SchemaType)
+						{
+						case SCHEMA_Data:
+							Writer.Printf("unreal.generated.{0}.subobjects.{1},", ActorClassName.ToLower(), ActorSubObjectName);
+							break;
+						case SCHEMA_OwnerOnly:
+							Writer.Printf("unreal.generated.{0}.subobjects.{1}OwnerOnly,", ActorClassName.ToLower(), ActorSubObjectName);
+							break;
+						case SCHEMA_Handover:
+							Writer.Printf("unreal.generated.{0}.subobjects.{1}Handover,", ActorClassName.ToLower(), ActorSubObjectName);
+							break;
+						default:
+							break;
+						}
+					}
+				});
+			}
+		}
+		// Dynamic subobjects.
+		for (const auto& GeneratedSubObjectClass : SchemaDatabase->SubobjectClassPathToSchema)
+		{
+			const FString& SubObjectClassName = UnrealNameToSchemaComponentName(GeneratedSubObjectClass.Value.GeneratedSchemaName);
+			for (auto SubObjectNumber = 0; SubObjectNumber < GeneratedSubObjectClass.Value.DynamicSubobjectComponents.Num();
+				 ++SubObjectNumber)
+			{
+				const FDynamicSubobjectSchemaData& SubObjectSchemaData =
+					GeneratedSubObjectClass.Value.DynamicSubobjectComponents[SubObjectNumber];
+				ForAllSchemaComponentTypes([&](ESchemaComponentType SchemaType) {
+					const Worker_ComponentId& ComponentId = SubObjectSchemaData.SchemaComponents[SchemaType];
+					if (ComponentId != 0)
+					{
+						switch (SchemaType)
+						{
+						case SCHEMA_Data:
+							Writer.Printf("unreal.generated.{0}Dynamic{1},", SubObjectClassName, SubObjectNumber + 1);
+							break;
+						case SCHEMA_OwnerOnly:
+							Writer.Printf("unreal.generated.{0}OwnerOnlyDynamic{1},", SubObjectClassName, SubObjectNumber + 1);
+							break;
+						case SCHEMA_Handover:
+							Writer.Printf("unreal.generated.{0}HandoverDynamic{1},", SubObjectClassName, SubObjectNumber + 1);
+							break;
+						default:
+							break;
+						}
+					}
+				});
+			}
+		}
+	}
+
+	Writer.RemoveTrailingComma();
+
+	Writer.Outdent().Print("];");
+	Writer.Outdent().Print("}");
+
+	Writer.WriteToFile(FString::Printf(TEXT("%sComponentSets/ServerAuthoritativeComponentSet.schema"), *SchemaOutputPath));
+}
+
+void WriteComponentSetBySchemaType(const USchemaDatabase* SchemaDatabase, ESchemaComponentType SchemaType)
```

Comment:
nit - lots of duplication with the above function - would be good to have a function that just took a schema mask/filter and applied it